django/contrib/staticfiles/storage.py

```diff
@@ -163,7 +163,7 @@ def converter(matchobj):
             matched, url = matchobj.groups()
 
             # Ignore absolute/protocol-relative, fragments and data-uri URLs.
-            if url.startswith(('http:', 'https:', '//', '#', 'data:')):
+            if re.match(r'^(?i)([a-z]+:|//|#)', url):
```

Comment:
Missing a test that fails if `(?i)` is removed. Actually as little as `r'^[a-z]+:'` will allow the tests to pass.